static/app/views/projectDetail/projectLatestAlerts.tsx

```diff
@@ -1,196 +1,135 @@
+              date={dateStarted}
+              tooltipUnderlineColor={getStatusColor(statusProps)}
+            />
+          )}
+        </AlertDate>
+      </AlertDetails>
+    </AlertRowLink>
+  );
+}
+
+interface ProjectLatestAlertsProps {
   isProjectStabilized: boolean;
   location: Location;
   organization: Organization;
   projectSlug: string;
-};
-
-type State = {
-  resolvedAlerts: Incident[] | null;
-  unresolvedAlerts: Incident[] | null;
-  hasAlertRule?: boolean;
-} & DeprecatedAsyncComponent['state'];
-
-class ProjectLatestAlerts extends DeprecatedAsyncComponent<Props, State> {
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
-    const {location, isProjectStabilized} = this.props;
-    // TODO(project-detail): we temporarily removed refetching based on timeselector
-    if (
-      this.state !== nextState ||
-      didProjectOrEnvironmentChange(location, nextProps.location) ||
-      isProjectStabilized !== nextProps.isProjectStabilized
-    ) {
-      return true;
-    }
-
-    return false;
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const {location, isProjectStabilized} = this.props;
-
-    if (
-      didProjectOrEnvironmentChange(prevProps.location, location) ||
-      prevProps.isProjectStabilized !== isProjectStabilized
-    ) {
-      this.remountComponent();
-    }
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {location, organization, isProjectStabilized} = this.props;
-
-    if (!isProjectStabilized) {
-      return [];
-    }
-
-    const query = {
-      ...pick(location.query, Object.values(URL_PARAM)),
-      per_page: 3,
-    };
-
-    // we are listing 3 alerts total, first unresolved and then we fill with resolved
-    return [
-      [
-        'unresolvedAlerts',
-        `/organizations/${organization.slug}/incidents/`,
-        {query: {...query, status: 'open'}},
-      ],
-      [
-        'resolvedAlerts',
-        `/organizations/${organization.slug}/incidents/`,
-        {query: {...query, status: 'closed'}},
-      ],
-    ];
-  }
-
-  /**
-   * If our alerts are empty, determine if we've configured alert rules (empty message differs then)
-   */
-  async onLoadAllEndpointsSuccess() {
-    const {unresolvedAlerts, resolvedAlerts} = this.state;
-    const {location, organization, isProjectStabilized} = this.props;
-
-    if (!isProjectStabilized) {
-      return;
-    }
-
-    if ([...(unresolvedAlerts ?? []), ...(resolvedAlerts ?? [])].length !== 0) {
-      this.setState({hasAlertRule: true});
-      return;
-    }
-
-    this.setState({loading: true});
+}
 
-    const alertRules = await this.api.requestPromise(
+function ProjectLatestAlerts({
+  location,
+  organization,
+  isProjectStabilized,
+  projectSlug,
+}: ProjectLatestAlertsProps) {
+  const query = {
+    ...pick(location.query, Object.values(URL_PARAM)),
+    per_page: 3,
+  };
+  const {data: unresolvedAlerts = [], isLoading: unresolvedAlertsIsLoading} = useApiQuery<
```

Comment:
We should be using the `isError` state here right? Need some sort of error handling

Reply:
ah yes you're right